src/mqt/bench/evaluation/evaluation.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import pickle
+import sys
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING or sys.version_info >= (3, 10, 0):  # pragma: no cover
+    pass
+else:
+    pass
+
+from pathlib import Path
+
+import numpy as np
+from joblib import Parallel, delayed
+from mqt.bench import utils
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.transpiler.passes import RemoveBarriers
+
+if TYPE_CHECKING or sys.version_info >= (3, 10, 0):  # pragma: no cover
+    from importlib import resources
+else:
+    import importlib_resources as resources
+
+
+def create_statistics() -> None:
+    source_circuits_list = [file for file in Path(utils.get_qasm_output_path()).iterdir() if file.suffix == ".qasm"]
+
+    res = Parallel(n_jobs=-1, verbose=100)(delayed(evaluate_qasm_file)(filename) for filename in source_circuits_list)
+    target_dir = Path(resources.files("mqt.bench") / "evaluation/")
+    with Path(target_dir / "evaluation_data.pkl").open("wb") as f:
+        pickle.dump(res, f)
+
+
+def evaluate_qasm_file(filename: str) -> tuple[str, int, int, int, int, float, float, float, float, float]:
```

Comment:
How about making that return type a `TypedDict` so that the individual evaluation metrics also get a meaning assigned to them?

Reply:
Good point, implemented such a `TypedDict`.